lib/SPIRV/LLVMToSPIRVDbgTran.cpp

```diff
@@ -1150,9 +1151,14 @@ SPIRVEntry *LLVMToSPIRVDbgTran::transDbgFunction(const DISubprogram *Func) {
         SPIRVValue *SPIRVFunc = SPIRVWriter->getTranslatedValue(&F);
         assert(SPIRVFunc && "All function must be already translated");
         Ops[FunctionIdIdx] = SPIRVFunc->getId();
+        FuncDef = SPIRVFunc;
         break;
       }
     }
+    // For NonSemantic.Shader.DebugInfo we store Function Id index as a
+    // separate DebugFunctionDefinition instruction.
+    if (isNonSemanticDebugInfo())
+      Ops.pop_back();
```

Comment:
I guess, because of SPIRVToLLVMDbgTran a cleaner solution of removing FunctionIdIdx from SPIRV.debug.h is quite hard. Yet, I'd suggest to a little improvement here: move resize of the vector to line 1147 and do `for (const llvm::Function &F : M->functions()) {` loop is else branch.

Reply:
I'm wrong